projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathAssertion.java

```diff
@@ -0,0 +1,77 @@
+package org.batfish.question.jsonpath;
+
+import static org.batfish.question.jsonpath.JsonPathAssertionType.count;
+import static org.batfish.question.jsonpath.JsonPathAssertionType.equal;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.batfish.common.BatfishException;
+import org.batfish.common.util.BatfishObjectMapper;
+
+public class JsonPathAssertion {
+
+  private static final String PROP_TYPE = "type";
+
+  private static final String PROP_EXPECT = "expect";
+
+  private JsonPathAssertionType _assertionType;
+
+  private JsonNode _expect;
+
+  public boolean evaluate(ArrayNode suffixes) {
+    switch (getType()) {
+    case count:
+      if (_expect.isInt()) {
```

Comment:
I prefer to write these with exit paths early, business logic late. It's easier for a reader to confirm properties like "we don't fall through the case":

```java
if (!_expect.isInt()) {
  throw...
}

return suffixes.size() == _expect.asInt();
```